notify/impl.go

```diff
@@ -1181,36 +1202,36 @@ func (n *VictorOps) Notify(ctx context.Context, as ...*types.Alert) (bool, error
 		level.Debug(n.logger).Log("msg", "Truncated stateMessage due to VictorOps stateMessage limit", "truncated_state_message", stateMessage, "incident", key)
 	}
 
-	msg := &victorOpsMessage{
-		MessageType:       messageType,
-		EntityID:          hashKey(key),
-		EntityDisplayName: tmpl(n.conf.EntityDisplayName),
-		StateMessage:      stateMessage,
-		MonitoringTool:    tmpl(n.conf.MonitoringTool),
+	msg := map[string]string{
+		"message_type":        messageType,
+		"entity_id":           hashKey(key),
+		"entity_display_name": tmpl(n.conf.EntityDisplayName),
+		"state_message":       stateMessage,
+		"monitoring_tool":     tmpl(n.conf.MonitoringTool),
 	}
 
 	if err != nil {
-		return false, fmt.Errorf("templating error: %s", err)
+		return nil, fmt.Errorf("templating error: %s", err)
 	}
 
-	var buf bytes.Buffer
-	if err := json.NewEncoder(&buf).Encode(msg); err != nil {
-		return false, err
-	}
+	for k, v := range n.conf.CustomFields {
 
-	c, err := commoncfg.NewClientFromConfig(*n.conf.HTTPConfig, "victorops")
-	if err != nil {
-		return false, err
+		//Validate if the custom field is not one of the fixed fields above.
+		if _, ok := msg[k]; !ok {
```

Comment:
just a suggestion but it reads more easily IMO like this:

```go
if _, ok := msg[k]; ok {
  level.Debug(n.logger).Log("msg", "Ignoring custom field %q as it is already defined as a fixed field", "omitted_field", k, "incident", key)
  continue
}
msg[k] = tmpl(v)
...
```

Reply:
Thank you. I do agree this looks cleaner